### PR TITLE
feat: implement external process triggering

### PR DESCRIPTION
**Pull request by rauno56**
A generic way of letting 3rd parties trigger processes like signing up for 3rd pillar fund in Tuleva.

There's no procedures defined right now so we just log a message in the end of the initialization process.

### CODE REVIEW

File: src/components/TriggerProcedure/externalProviderUtils.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+const EXTERNAL_AUTHENTICATOR_PROVIDER = 'EXTERNAL_AUTHENTICATOR_PROVIDER';
+const EXTERNAL_AUTHENTICATOR_REDIRECT_URI = 'EXTERNAL_AUTHENTICATOR_REDIRECT_URI';
+
+enum ExternalProvider {
+  PROVIDER1 = 'provider1',
+  PROVIDER2 = 'provider2',
```
Comment (rauno56) [inline src/components/TriggerProcedure/externalProviderUtils.ts:6]: Currently defined providers. We'll change them once we have providers signed up.

File: src/components/TriggerProcedure/externalProviderUtils.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  }
+  throw new Error('Redirect URI missing');
+};
+
+const validateHandoverToken = (value?: unknown) => {
+  if (value && typeof value === 'string') {
+    return value;
+  }
+  throw new Error(`Invalid handoverToken: ${value}`);
+};
```
Comment (rauno56) [inline src/components/TriggerProcedure/externalProviderUtils.ts:44]: Maybe convert that to use a validation library eventually. Didn't want to waste time with that for now.

File: src/components/TriggerProcedure/TriggerProcedure.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+  return (
+    <div
+      style={{
+        width: '100%',
+        height: '100vh',
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+      }}
```
Comment (ErkoRisthein) [inline src/components/TriggerProcedure/TriggerProcedure.tsx:64]: Any way to use bootstrap classes for this?

File: src/components/TriggerProcedure/externalProviderUtils.ts
Comment (ErkoRisthein) [inline src/components/TriggerProcedure/externalProviderUtils.ts:1]: Utils sounds like a violation of the SRP (Single Responsibility Principle). Does this component do too many things? We should split it up into smaller logical components then.

File: src/components/TriggerProcedure/TriggerProcess.test.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+          ctx.json({
+            accessToken: '_accessToken_',
+          }),
+        );
+      }),
+    );
+    consoleError = jest.spyOn(console, 'error').mockImplementation(() => {});
+  });
+
+  describe('... and invalid data is provider', () => {
```
Comment (ErkoRisthein) [inline src/components/TriggerProcedure/TriggerProcess.test.tsx:56]: provided?

File: src/components/common/api.ts [hunk 5]
```diff
@@ -350,6 +350,13 @@ export function redirectToPayment(payment: Payment, token: string): void {
   });
 }
 
+// Exchanges handoverToken for accessToken and overwrites the session
+export function exchangeHandoverToken(handoverToken: string): Promise<{ accessToken: string }> {
```
[the remark this answers is not in the document]
Comment (rauno56) [reply src/components/common/api.ts:354]: Yes! I just followed, what seemed to be, a convention, but I agree that it belong to under the component just as well if not better.

File: src/components/TriggerProcedure/TriggerProcedure.tsx (hunk 3 above)
Comment (rauno56) [reply src/components/TriggerProcedure/TriggerProcedure.tsx:64]: 1. We'd need to set width and height anyway
2. I don't think there's utility classes for vertical alignment

We could turn `justifyContent` to `text-center` if we turned the text inside into a `<p>`, but I have hard time to "justify" that.

File: src/components/TriggerProcedure/externalProviderUtils.ts
Comment (rauno56) [reply src/components/TriggerProcedure/externalProviderUtils.ts:1]: It's really a file of misc utilities that have no deps to reduce noise in the main component. Similar to `comomn/utils.js`, or `dateUtils.js`. Spliting them into different files increases pain of maintaining and finding them.

File: src/components/common/api.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (rauno56) [reply src/components/common/api.ts:354]: On another thought... `post` and `getEndpoint` are utilities only used in `common/api.ts`. Refactoring means, I need to import something that might be considered an internal API into another external module.

I'm still for making the move though.

**Review by ErkoRisthein — APPROVED** [review]

**Pull request merged**